Cards/CardBase.gd

```diff
@@ -0,0 +1,84 @@
+extends Resource
+class_name CardBase
+## Resource that holds data about a card and provides functions that all derived cards could
+## use if they want.
+##
+## The intention with this is to provide functionality that is common to a lot of cards,
+## like dealing damage, drawing cards, and restoring resources.
+## If you want functionality that's unique to a certain card, then create a new child of this
+## and override one of the functions below.
+## This resource also has data that is used for displaying the card in the world, like
+## description, title, and key art.
+## This includes functionality for applying damage to both the victim and attacker. Attackers may
+## wish to take damage in some contexts.
+## For example, consider the card: "Deal 10 damage to all enemies, but take 3 damage"
+
+
+@export var damage_to_apply_to_victim: float = 0.0
+@export var damage_to_apply_to_attacker: float = 0.0
+@export var buffs_to_apply_to_victim: Array[BuffBase]
+@export var buffs_to_apply_to_attacker: Array[BuffBase]
+@export var affect_all_victims: bool = false
+@export var affect_all_attackers: bool = false
+@export var application_type: Enums.ApplicationType = Enums.ApplicationType.ENEMY_ONLY
+@export var card_title: String = "NULL"
+@export var card_key_art: ImageTexture = null
+@export var card_description: String = "NULL"
+
+
+func try_play_card(attacker: Entity, victim: Entity) -> bool:
```

Comment:
It is a bit weird when you consider all the cases, but since the system is capable of it I figured we should allow it. Never know what weird card designs we may come up with in the future.

I'm open to different naming suggestions for attacker/victim. That was the most clear one I could come up with

Reply:
Maybe caster / target would work better for the name ?

Reply:
Caster/Target sounds pretty clear and should work for all cases I believe ? 
@Turtyo what do you think, should we just do it as part of this one to get it out of the way ?

Reply:
There a lot of instances of attacker / victim within the code. On one hand, it might take some time to do, but since it's also in a lot of files, if we don't do it now it might lead to merge conflicts / wrong naming.
I think it should be done now even if it takes a bit of time.

Reply:
I can do it now, should be a quick find+replace